File: examples/python/instagram.py
import requests
import os

API_KEY = os.getenv('COGNIMA_API_KEY', 'ck_your_api_key')
BASE_URL = 'https://cog2.cognima.com.br/api/v1'

def download_instagram():
    url = f'{BASE_URL}/instagram/download'
    headers = {
        'X-API-Key': API_KEY,
        'Content-Type': 'application/json'
    }
    data = {
        'url': 'https://www.instagram.com/p/ABC123xyz/'
    }
    
    response = requests.post(url, json=data, headers=headers)
    
    if response.status_code == 200:
        result = response.json()
        print('Instagram Post Info:')
        print(f"Type: {result['data']['type']}")
        print(f"Caption: {result['data']['caption']}")
        print(f"Author: {result['data']['author']['username']}")
        print(f"Likes: {result['data']['likes']:,}")
        print(f"\nMedia URLs:")
        for i, media in enumerate(result['data']['media'], 1):
            print(f"{i}. {media['type']}: {media['url']}")
    else:
        print(f"Error {response.status_code}: {response.json()}")

if __name__ == '__main__':
    download_instagram()

File: examples/nodejs/instagram.js
const axios = require('axios');

const API_KEY = process.env.COGNIMA_API_KEY || 'ck_your_api_key';
const BASE_URL = 'https://cog.api.br/api/v1';

async function downloadInstagram() {
  try {
    const response = await axios.post(
      `${BASE_URL}/instagram/download`,
      { url: 'https://www.instagram.com/p/ABC123xyz/' },
      {
        headers: {
          'X-API-Key': API_KEY,
          'Content-Type': 'application/json'
        }
      }
    );
    
    const data = response.data.data;
    console.log('Instagram Post Info:');
    console.log(`Type: ${data.type}`);
    console.log(`Caption: ${data.caption}`);
    console.log(`Author: ${data.author.username}`);
    console.log(`Likes: ${data.likes.toLocaleString()}`);
    console.log('\nMedia URLs:');
    data.media.forEach((media, i) => {
      console.log(`${i + 1}. ${media.type}: ${media.url}`);
    });
  } catch (error) {
    console.error('Error:', error.response?.data || error.message);
  }
}

downloadInstagram();
